recommender/recommender.py

```diff
@@ -235,98 +246,119 @@ def error_handler(self, error_msg, event, resource_id=None):
-                data['id']: the ID of the resouce which was upvoted
-        Returns:
-                result: dict in JSON format
-                result['Success']: the boolean indicator for whether the process of this voting action is complete
-                result['error']: the error message generated when the process fails
-                result['oldVotes']: the original votes
-                result['newVotes']: the votes after this action
-                result['toggle']: the boolean indicator for whether the resource was switched from downvoted to upvoted
+                data['DISABLE_DEV_UX']: whether to disable the UX under development
+                data['ENTRIES_PER_PAGE']: the number of resources in each page
+                data['PAGE_SPAN']: page range in pagination control
+                data['INTRO']: whether to take users on a little tour when they see the RecommenderXBlock first time
         """
-        resource_id = data['id']
-        idx = self.get_entry_index(resource_id, self.recommendations)
-        if idx not in range(0, len(self.recommendations)):
-            msg = 'The selected resource is not existing'
-            return self.error_handler(msg, 'recommender_upvote', resource_id)
+        if data['DISABLE_DEV_UX'].lower() == 'false':
+            self.client_side_settings['DISABLE_DEV_UX'] = False
+        else:
+            self.client_side_settings['DISABLE_DEV_UX'] = True
 
-        result = {}
-        result['id'] = resource_id
-        result['oldVotes'] = (self.recommendations[idx]['upvotes'] -
-                              self.recommendations[idx]['downvotes'])
-        if resource_id in self.upvoted_ids:
-            del self.upvoted_ids[self.upvoted_ids.index(resource_id)]
-            self.recommendations[idx]['upvotes'] -= 1
-            result['newVotes'] = (self.recommendations[idx]['upvotes'] -
-                                  self.recommendations[idx]['downvotes'])
-            result['Success'] = True
-            tracker.emit('recommender_upvote', result)
-            return result
+        if data['INTRO'].lower() == 'false':
+            self.intro_enabled = False
+        else:
+            self.intro_enabled = True
 
-        if resource_id in self.downvoted_ids:
-            del self.downvoted_ids[self.downvoted_ids.index(resource_id)]
-            self.recommendations[idx]['downvotes'] -= 1
-            result['toggle'] = True
-        self.upvoted_ids.append(resource_id)
-        self.recommendations[idx]['upvotes'] += 1
-        result['newVotes'] = (self.recommendations[idx]['upvotes'] -
-                              self.recommendations[idx]['downvotes'])
-        result['Success'] = True
-        tracker.emit('recommender_upvote', result)
-        return result
+        for key in ['PAGE_SPAN', 'ENTRIES_PER_PAGE']:
+            self.client_side_settings[key] = int(data[key])
+
+        tracker.emit('set_client_side_settings', data)
+        return {'Success': True}
```

Comment:
You shouldn't need to indicate success in the body -- just returning a response with status code 200 should be enough.